config/software/cleanup.rb

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright 2012-2016 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name "cleanup"
```

Comment:
I like this as an (optional) omnibus feature. Other things that are worth discussing would be a 'whitelist' type functionality for specific files/libs and the ability to disable this functionality based on platform/machine/regex/something.